core/domain/user_services.py

```diff
@@ -2821,3 +2821,44 @@ def log_username_change(committer_id, old_username, new_username):
     audit_models.UsernameChangeAuditModel(
         id=model_id, committer_id=committer_id, old_username=old_username,
         new_username=new_username).put()
+
+
+def create_login_url(target_url):
+    """Creates a login url.
+
+    Args:
+        target_url: str. The URL to redirect to after login.
+
+    Returns:
+        str. The correct login URL that includes the page to redirect to.
+    """
+
+    return current_user_services.create_login_url(target_url)
+
+
+def is_current_user_super_admin():
+    """Checks whether the current logged user is super admin.
+
+    Returns:
+        bool. Whether the current logged user is super admin.
+    """
+
+    return current_user_services.is_current_user_super_admin()
+
+
+def get_current_gae_id():
+    """Gets the GAE ID of current user.
```

Comment:
I mean we use GAE ID primarily in base.py https://github.com/oppia/oppia/blob/f2e6d7f2bf286a8eec79dbd006b96452c819e1c1/core/controllers/base.py#L183-L199. I'm not sure if I want to do any bigger refactor here since Brian refactoring the whole authentication system that will probably also affect this part.

Reply:
Unless I'm mistaken, a large refactor isn't needed, is it? From what I can tell, the only operations we care about are:
1. Is there a gae_id (we don't technically care what it is at the controller level, only that we can use it to fetch user settings). This is equivalent to "is there a current user logged in"
2. Retrieve the current user's settings

I'm proposing that we avoid needing to expose GAE ID here by replacing ``get_current_gae_id`` with two targeted functions to satisfy each of the above: ``is_a_user_currently_logged_in`` and ``get_current_user_settings`` (I suppose we could actually combine these into one since we already return None for ``get_user_settings_by_gae_id`` but I defer to you here).

@brianrodri do you have any specific thoughts on this? I'm not keen on exposing GAE ID at all since it seems principally wrong, but I understand that there are a lot of shifting pieces here at the moment.

Reply:
While I agree that we should prefer hiding the GAE ID, I don't think this PR is the right place to do it. I'm fine with merging as is and fine-tuning the functions you propose during the migration to the new authentication stuff.

WDYT?

Reply:
Will the authentication work actually change this codepath? If so, I agree with you. If not, I think we should just make the change here since it seems really straightforward (unless I'm missing something, this looks like a ~10 line change that's effectively a refactor and nothing else). Will our handling of GAE IDs change with the new auth stuff @brianrodri?

Reply:
Ah sorry for not making that clear, yes this will code will be updated in the authentication changes. Given this, it sounds like we can resolve this thread and continue discussions in the design doc.

Reply:
Thanks for clarifying @brianrodri.

@vojtechjelinek can you add a TODO here on the tracking issue for the GAE user migration (@brianrodri what issue number should we use?) or file a new one? Given that this isn't an ideal pattern to keep, we should make sure it's being tracked so that we do address this even if it's in a follow-up PR.

Reply:
Added TODO to the code in base.py

Reply:
Thanks!